pymeasure/instruments/racal/racal1992.py

```diff
@@ -0,0 +1,360 @@
+    """ Represents the Racal-Dana 1992 Universal counter
+
+    .. code-block:: python
+
+        from pymeasure.instruments.racal import Racal1992
+        counter = Racal1992("GPIB0::10")
+
+    This class should also work for Racal-Dana 1991, it has the same
+    product manual, as long as you don't use functionality that requires
+    channel B.
+    """
+
+    int_types = ['SF', 'RS', 'UT', 'MS', 'TA']
+    float_types = ['CK', 'FA', 'PA', 'TI', 'PH', 'RA',
+                   'MX', 'MZ', 'LA', 'LB', 'FC', 'RC', 'DT', 'GS']
+
+    channel_params = {
+        'A' : {                                                         # noqa
+            'coupling'      : { 'AC'   : 'AAC',  'DC'     : 'ADC' },    # noqa
+            'attenuation'   : { 'X1'   : 'AAD',  'X10'    : 'AAE' },    # noqa
+            'trigger'       : { 'auto' : 'AAU',  'manual' : 'AMN' },    # noqa
+            'impedance'     : { '50'   : 'ALI',  '1M'     : 'AHI' },    # noqa
+            'slope'         : { 'pos'  : 'APS',  'neg'    : 'ANS' },    # noqa
+            'filtering'     : { True   : 'AFE',  False    : 'AFD' },    # noqa
+            'trigger_level' : None,                                     # noqa
+            },
+        'B' : {                                                         # noqa
+            'coupling'      : { 'AC'        : 'BAC',  'DC'     : 'BDC' },    # noqa
+            'attenuation'   : { 'X1'        : 'BAD',  'X10'    : 'BAE' },    # noqa
+            'trigger'       : { 'auto'      : 'BAU',  'manual' : 'BMN' },    # noqa
+            'impedance'     : { '50'        : 'BLI',  '1M'     : 'BHI' },    # noqa
+            'slope'         : { 'pos'       : 'BPS',  'neg'    : 'BNS' },    # noqa
+            'input_select'  : { 'separate'  : 'BCS',  'common' : 'BCC' },    # noqa
+            'trigger_level' : None,                                     # noqa
+            },
+    }
+
+    operating_modes = {
+        'self_check'      : 'CK',    # noqa
+        'frequency_a'     : 'FA',    # noqa
+        'period_a'        : 'PA',    # noqa
+        'phase_a_rel_b'   : 'PH',    # noqa
+        'ratio_a_to_b'    : 'RA',    # noqa
+        'ratio_c_to_b'    : 'RC',    # noqa
+        'interval_a_to_b' : 'TI',    # noqa
+        'total_a_by_b'    : 'TA',    # noqa
+        'frequency_c'     : 'FC',    # noqa
+    }
+
+    def decode(v, allowed_types=None):
+        """All values returned follow the same format: 2 letters to indicate
+        the type of the value returned, followed by a floating point number
+        (which could be an integer, of course.)
+        This here, for example, is math constant Z: MZ+001.00000000E+00
+        """
+        if len(v) != 19:
+            raise ReturnValueError("Length of instrument response must always be 19 characters")
+
+        val_type = v[0:2]
+        val = float(v[2:19])
+
+        if allowed_types and val_type not in allowed_types:
+            raise ValueError("Unexpected value type returned")
```

Comment:
Just a remark/suggestion: In general when throwing exceptions, I try to give the user information that could be useful for subsequent troubleshooting, e.g. the value/object that violated some expectation. E.g. here
```suggestion
            raise ValueError(f"Unexpected value type returned: '{val_type}'")
```
or above the response that is not 19 chars long, etc.